consumer/src/events.rs

```diff
@@ -677,4 +714,294 @@ impl Processor {
+        owner: Set(owner.to_string()),
+        mint: Set(mint.to_string()),
+        metadata: Set(metadata_pubkey.to_string()),
+        ..Default::default()
+    })
+    .await?;
+
+    producer
+        .send(
+            Some(&SolanaNftEvents {
+                event: Some(SolanaNftEvent::IndexExternalSolanaCollection(
+                    SolanaCollectionPayload {
+                        supply: collection.supply.map(|s| s.print_max_supply),
+                        mint_address: mint.to_string(),
+                        seller_fee_basis_points,
+                        creators,
+                        metadata: Some(Metadata {
+                            name: metadata.name,
+                            description: metadata.description,
+                            symbol: metadata.symbol,
+                            attributes,
+                            uri: collection.content.json_uri,
+                            image,
+                        }),
+                        files,
+                        update_authority: update_authority.to_string(),
+                    },
+                )),
+            }),
+            Some(&SolanaNftEventKey {
+                id: collection_model.id.to_string(),
+                project_id,
+                user_id,
+            }),
+        )
+        .await
+        .map_err(ProcessorErrorKind::SendError)?;
+
+    Ok(collection_model)
+}
+
+async fn emit_index_collection_mint_event(
+    project_id: String,
+    user_id: String,
+    collection: Uuid,
+    asset: Asset,
+    producer: Producer<SolanaNftEvents>,
+) -> ProcessResult<collection_mints::ActiveModel> {
+    let owner = asset.ownership.owner.into();
+
+    let mint = asset.id.into();
+    let ata = get_associated_token_address(&owner, &mint);
+    let seller_fee_basis_points = asset.royalty.basis_points;
+    let metadata = asset.content.metadata;
+    let update_authority = asset
+        .authorities
+        .get(0)
+        .ok_or(ProcessorErrorKind::IndexNotFound)?
+        .address
+        .clone();
+
+    let files = asset
+        .content
+        .files
+        .map(|fs| fs.iter().map(Into::into).collect::<Vec<File>>())
+        .unwrap_or_default();
+
+    let image = files
+        .iter()
+        .find(|f| f.mime.as_ref().map_or(false, |m| m.contains("image")))
+        .map(|f| f.uri.clone())
+        .unwrap_or_default();
+
+    let attributes = metadata
+        .attributes
+        .map(|attributes| attributes.iter().map(Into::into).collect())
+        .unwrap_or_default();
+
+    let creators = asset
+        .creators
+        .iter()
+        .map(|c| SolanaCreator {
+            address: c.address.to_string(),
+            verified: c.verified,
+            share: c.share,
+        })
+        .collect::<Vec<_>>();
+
+    #[allow(clippy::cast_sign_loss)]
+    let uuid = Uuid::from_u64_pair(Utc::now().timestamp_nanos() as u64, rand::random::<u64>());
```

Comment:
Since addresses are unique why not use it as it as part of the pair?

Reply:
Address has 32 bytes so can only use first 8 bytes for the other half or 16 bytes for both parts so I think random number would be a better option?